File: robocode/robocode/control/RobotResults.java
/*******************************************************************************
 * Copyright (c) 2001, 2008 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Common Public License v1.0
 * which accompanies this distribution, and is available at
 * http://robocode.sourceforge.net/license/cpl-v10.html
 *
 * Contributors:
 *     Mathew A. Nelson
 *     - Initial API and implementation
 *     Flemming N. Larsen
 *     - Changed to be consistent with the battle results and ranking scores
 *     - This class now implements java.io.Serializable
 *     - Updated Javadocs
 *******************************************************************************/
package robocode.control;

import robocode.BattleResults;


/**
 * Contains the battle results for an individual robot, which is given as input
 * parameter with the
 * {@link RobocodeListener#battleComplete(BattleSpecification, RobotResults[])
 * RobocodeListener#battleComplete()} event handler.
 *
 * @author Mathew A. Nelson (original)
 * @author Flemming N. Larsen (contributor)
 * @see RobocodeListener
 */
public class RobotResults extends BattleResults {

	private static final long serialVersionUID = 2L;

	private RobotSpecification robot;

	/**
	 * Constructs a new RobotResults.
	 *
	 * @param robot             the robot these results are for
	 * @param teamLeaderName    team name
     * @param rank              the rank of the robot in the battle
	 * @param score             the total score for the robot in the battle
	 * @param survival          the survival score for the robot in the battle
	 * @param lastSurvivorBonus the last survivor bonus for the robot in the battle
	 * @param bulletDamage      the bullet damage score for the robot in the battle
	 * @param bulletDamageBonus the bullet damage bonus for the robot in the battle
	 * @param ramDamage         the ramming damage for the robot in the battle
	 * @param ramDamageBonus    the ramming damage bonus for the robot in the battle
	 * @param firsts            the number of rounds this robot placed first
	 * @param seconds           the number of rounds this robot placed second
	 * @param thirds            the number of rounds this robot placed third
	 */
	public RobotResults(
			RobotSpecification robot,
            String teamLeaderName,
			int rank,
			double score,
			double survival,
			double lastSurvivorBonus,
			double bulletDamage,
			double bulletDamageBonus,
			double ramDamage,
			double ramDamageBonus,
			int firsts,
			int seconds,
			int thirds
			) {
        super(teamLeaderName, rank, score, survival, lastSurvivorBonus, bulletDamage, bulletDamageBonus, ramDamage, ramDamageBonus, firsts, seconds, thirds);
        this.robot = robot;
	}

    public RobotResults(
            RobotSpecification robot,
            BattleResults results) {
        super(results.getTeamLeaderName(), results.getRank(), results.getScore(), results.getSurvival(), results.getLastSurvivorBonus(),
                results.getBulletDamage(), results.getBulletDamageBonus(), results.getRamDamage(), results.getRamDamageBonus(),
                results.getFirsts(), results.getSeconds(), results.getThirds());
        this.robot = robot;
    }

	/**
	 * Returns the robot these results are meant for.
	 *
	 * @return the robot these results are meant for.
	 */
	public RobotSpecification getRobot() {
		return robot;
	}
}

File: plugins/dotnet/robocode.dotnet.napi/src/control/RobotResults.cs
using System.Runtime.CompilerServices;
using IKVM.Attributes;

namespace robocode.control
{
    public class RobotResults : BattleResults
    {
        private const long serialVersionUID = 2L;
        [Modifiers(Modifiers.Private | Modifiers.Final)] private RobotSpecification robot;

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x1f, 0xdf, 0x38, 0x67})]
        public RobotResults(RobotSpecification robot, BattleResults results)
            : base(
                results.getTeamLeaderName(), results.getRank(), results.getScore(), results.getSurvival(),
                results.getLastSurvivorBonus(), results.getBulletDamage(), results.getBulletDamageBonus(),
                results.getRamDamage(), results.getRamDamageBonus(), results.getFirsts(), results.getSeconds(),
                results.getThirds())
        {
            this.robot = robot;
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x10, 0x9f, 6, 0x67})]
        public RobotResults(RobotSpecification robot, string teamLeaderName, int rank, double score, double survival,
                            double lastSurvivorBonus, double bulletDamage, double bulletDamageBonus, double ramDamage,
                            double ramDamageBonus, int firsts, int seconds, int thirds)
            : base(
                teamLeaderName, rank, score, survival, lastSurvivorBonus, bulletDamage, bulletDamageBonus, ramDamage,
                ramDamageBonus, firsts, seconds, thirds)
        {
            this.robot = robot;
        }

        [LineNumberTable(new byte[] {0x37, 0x88, 0x6a, 0x2b, 0xa6})]
        public static RobotResults[] convertResults(BattleResults[] results)
        {
            var resultsArray = new RobotResults[results.Length];
            for (int i = 0; i < results.Length; i++)
            {
                resultsArray[i] = (RobotResults) results[i];
            }
            return resultsArray;
        }

        public virtual RobotSpecification getRobot()
        {
            return robot;
        }
    }
}